src/brpc/policy/baidu_rpc_protocol.cpp

```diff
@@ -80,11 +80,11 @@ static void SerializeRpcHeaderAndMeta(
         ::google::protobuf::io::CodedOutputStream coded_out(&arr_out);
         meta.SerializeWithCachedSizes(&coded_out); // not calling ByteSize again
         CHECK(!coded_out.HadError());
-        out->append(header_and_meta, sizeof(header_and_meta));
+        CHECK_EQ(0, out->append(header_and_meta, sizeof(header_and_meta)));
```

Comment:
IIUC this is just a assertion and will only be effective in debug build. Perhaps we should change the return type of `SerializeRpcHeaderAndMeta` into `bool` and handle it properly, similar to what we did for `SerializeAsCompressedData`.

Reply:
// The CHECK(condition) macro is active in both debug and release builds and
// effectively performs a LOG(FATAL) which terminates the process and
// generates a crashdump unless a debugger is attached.

https://github.com/apache/incubator-brpc/blob/master/src/butil/logging.h#L122

Reply:
Ok... It's interesting to see a release assertion macro) However, this error is recoverable. Resorting to `std::terminate` seems to be an overkill.

Reply:
LGTM